### PR TITLE
Adding Issuer DN checking

### PR DESCRIPTION
**Pull request by bvandersloot**

### CODE REVIEW

File: ztools/x509/x509.go [hunk 1]
```diff
@@ -778,6 +778,10 @@ func (c *Certificate) CheckSignatureFrom(parent *Certificate) (err error) {
 
 	// TODO(agl): don't ignore the path length constraint.
 
+	if parent.Subject.String() != c.Issuer.String() {
+		return errors.New("Mis-match issuer/subject")
+	}
+
```
Comment (zakird) [inline ztools/x509/x509.go]: This probably doesn't belong in this function. Your signature can be valid even with a issuer-subject mismatch.

Comment (zakird) [inline ztools/x509/x509.go]: Actually I take this back. While this function is named *terribly*, there _are_ other checks in it already. :(

File: ztools/zct/x509/x509.go [hunk 2]
```diff
@@ -619,6 +619,10 @@ func (c *Certificate) CheckSignatureFrom(parent *Certificate) (err error) {
 
 	// TODO(agl): don't ignore the path length constraint.
 
+	if parent.Subject.String() != c.Issuer.String() {
+		return errors.New(fmt.Sprintf("Mis-match issuer/subject (%s!=%s)", parent.Subject.String(), c.Issuer.String()))
+	}
+
```
Comment (zakird) [inline ztools/zct/x509/x509.go]: We can just check SubjectDN and IssuerDN instead of doing this twice.

Comment (bvandersloot) [inline ztools/zct/x509/x509.go]: Those are not parsed out in this version of x509.

**Pull request merged**